edb/server/dbview/dbview.pyx

```diff
@@ -182,22 +182,18 @@ cdef class Database:
         self._sql_to_compiled.clear()
         self._index.invalidate_caches()
 
-    cdef _cache_compiled_query(
-        self, key, compiled: dbstate.QueryUnitGroup, schema_version
-    ):
+    cdef _cache_compiled_query(self, key, compiled: dbstate.QueryUnitGroup):
         # `dbver` must be the schema version `compiled` was compiled upon
         assert compiled.cacheable
 
-        existing, existing_ver = self._eql_to_compiled.get(key, DICTDEFAULT)
-        if existing is not None and existing_ver == self.schema_version:
-            # We already have a cached query for a more recent DB version.
+        if key in self._eql_to_compiled:
+            # We already have a cached query for the current user schema
             return
 
-        # Store the matching schema version, see also the comments at origin
-        self._eql_to_compiled[key] = compiled, schema_version
+        self._eql_to_compiled[key] = compiled
         keys = []
         while self._eql_to_compiled.needs_cleanup():
-            query_req, (unit_group, _) = self._eql_to_compiled.cleanup_one()
+            query_req, unit_group = self._eql_to_compiled.cleanup_one()
             if len(unit_group) == 1:
                 keys.append(query_req.get_cache_key())
         if keys:
```

Comment:
TODO reminder: I think we need to defer these evictions until after the transaction completes.

The question of how to handle cache entries created during transactions is a tricky one in general, and will be more so once we switch back to functions, since it won't be visible to other tasks until it commits.
We might want to track explicitly whether it has been persisted succesfully yet?

Reply:
Yes, I'm thinking about something like what we do in dbviews - once applied successfully in DB, we replay the stuff in memory (`dbview` for in-tx cache and `database` for shared ones)